src/main/java/com/hierynomus/smbj/auth/NtlmAuthenticator.java

```diff
@@ -60,19 +61,24 @@ public NtlmAuthenticator create() {
         }
     }
 
-    public long authenticate(Connection connection, AuthenticationContext context) throws TransportException {
+    public Session authenticate(Connection connection, AuthenticationContext context) throws TransportException {
```

Comment:
It would be nicer to introduce a `SessionDetails` object which encapsulates the SigningKey and the SessionId. And leave the creation of the session to the connection. Now you creating a(nother) package cycle (I'm trying to break them at this moment).